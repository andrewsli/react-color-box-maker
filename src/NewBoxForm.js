import React, { Component } from "react";

class NewBoxForm extends Component {

  constructor(props) {
    super(props);
    this.state = { width: '', height: '', backgroundColor: '' };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(evt) {
    this.setState({ [evt.target.name]: evt.target.value });
  }

  //calls addBox parent function with submitted input, reset state
  handleSubmit(evt) {
    evt.preventDefault();
    this.props.addNewBox({
      width: `${this.state.width}px`,
      height: `${this.state.height}px`,
      backgroundColor: this.state.backgroundColor
    });
    this.setState({ width: '', height: '', backgroundColor: '' });
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <label htmlFor="height">Height (in px):</label>
        <input name="height" id="height" value={this.state.height} onChange={this.handleChange} />

        <label htmlFor="width">Width (in px):</label>
        <input name="width" id="width" value={this.state.width} onChange={this.handleChange} />

        <label htmlFor="backgroundColor">BackgroundColor:</label>
        <input name="backgroundColor" id="backgroundColor" value={this.state.backgroundColor} onChange={this.handleChange} />
        <button>Add box!</button>
      </form>
    )
  }
}

export default NewBoxForm;