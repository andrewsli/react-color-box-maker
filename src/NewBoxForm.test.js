import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import NewBoxForm from "./NewBoxForm";


it("renders using shallow", function () {
  shallow(<NewBoxForm />);
});


it("matches snapshot", function () {
  let wrapper = shallow(<NewBoxForm />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});

it("allows for changes in form inputs: height, width, and backgroundColor", function () {
  let wrapper = mount(<NewBoxForm />);
  //change height
  const heightInput = wrapper.find("#height");
  heightInput.instance().value = "100";
  heightInput.simulate("change");
  expect(wrapper.state().height).toEqual("100");
  //change width
  const widthInput = wrapper.find("#width");
  widthInput.instance().value = "100";
  widthInput.simulate("change");
  expect(wrapper.state().width).toEqual("100");
  //change backgroundColor
  const bgColorInput = wrapper.find("#backgroundColor");
  bgColorInput.instance().value = "green";
  bgColorInput.simulate("change");
  expect(wrapper.state().backgroundColor).toEqual("green");
})

it("runs a mocked function on submit", function () {
  const submitFn = jest.fn();
  let wrapper = mount(<NewBoxForm addNewBox={submitFn} />);
  const form = wrapper.find("form");

  form.simulate("submit");
  expect(submitFn).toHaveBeenCalled();
})

it("resets state on submission", function () {
  const submitFn = jest.fn();

  let wrapper = mount(<NewBoxForm addNewBox={submitFn} />);
  //get all our elements
  const heightInput = wrapper.find("#height");
  const widthInput = wrapper.find("#width");
  const bgColorInput = wrapper.find("#backgroundColor");
  const form = wrapper.find("form");

  //change all our fields and submit the form
  heightInput.instance().value = "100";
  heightInput.simulate("change");

  widthInput.instance().value = "100";
  widthInput.simulate("change");

  bgColorInput.instance().value = "green";
  bgColorInput.simulate("change");

  form.simulate("submit");

  expect(wrapper.state()).toEqual({ width: '', height: '', backgroundColor: '' });
})