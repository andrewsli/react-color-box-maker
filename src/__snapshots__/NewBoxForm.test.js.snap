// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    htmlFor="height"
  >
    Height (in px):
  </label>
  <input
    id="height"
    name="height"
    onChange={[Function]}
    value=""
  />
  <label
    htmlFor="width"
  >
    Width (in px):
  </label>
  <input
    id="width"
    name="width"
    onChange={[Function]}
    value=""
  />
  <label
    htmlFor="backgroundColor"
  >
    BackgroundColor:
  </label>
  <input
    id="backgroundColor"
    name="backgroundColor"
    onChange={[Function]}
    value=""
  />
  <button>
    Add box!
  </button>
</form>
`;
